gadget/validate.go

```diff
@@ -395,20 +395,25 @@ func ValidateContent(info *Info, gadgetSnapRootDir, kernelSnapRootDir string) er
 	// the gadget uses and as such there cannot be more than one
 	// such bootloader
 	for name, vol := range info.Volumes {
-		opts := &LayoutOptions{
-			GadgetRootDir: gadgetSnapRootDir,
-			KernelRootDir: kernelSnapRootDir,
-		}
-		// At this point we may not know what kernel will be used
-		// with the gadget yet. Skip this check in this case.
-		if kernelSnapRootDir == "" {
-			opts.SkipResolveContent = true
+		// Check that files shipped in the gadget have the expected sizes
+		for idx := range vol.Structure {
+			if err := checkGadgetContentImages(gadgetSnapRootDir, &vol.Structure[idx]); err != nil {
+				return err
+			}
 		}
-		lv, err := LayoutVolume(vol, nil, opts)
-		if err != nil {
-			return fmt.Errorf("invalid layout of volume %q: %v", name, err)
+		// Make sure that content can be resolved if the kernel snap is known.
+		if kernelSnapRootDir != "" {
+			kernelInfo, err := kernel.ReadInfo(kernelSnapRootDir)
+			if err != nil {
+				return err
+			}
```

Comment:
shouldn' this be done once outside the for loop and then check here for kernInfo != nil ?